pypdf/_writer.py

```diff
@@ -2421,17 +2421,34 @@ def pageMode(self, mode: PagemodeType) -> None:  # deprecated
         deprecation_with_replacement("pageMode", "page_mode", "3.0.0")
         self.page_mode = mode
 
-    def add_annotation(self, page_number: int, annotation: Dict[str, Any]) -> None:
+    def add_annotation(
+        self,
+        page_number: Union[int, PageObject],
+        annotation: Dict[str, Any],
+    ) -> DictionaryObject:
+        """
+        Add a single annotation to the page. Must be a new annotation (can not be recycled)
+
+        Args:
+            page: page object or number (used to be pagenumber : deprecated)
+            annotation : annotation to be added (created with annotation)
+
+        Returns:
+            the inserted object (to be used in pop-up creation argument for example)
+        """
+        page = page_number
+
+
         to_add = cast(DictionaryObject, _pdf_objectify(annotation))
-        to_add[NameObject("/P")] = self.get_object(self._pages)["/Kids"][page_number]  # type: ignore
-        page = self.pages[page_number]
+        to_add[NameObject("/P")] = page.indirect_reference
```

Comment:
This seems to cause issues:

```
>       to_add[NameObject("/P")] = page.indirect_reference
E       AttributeError: 'int' object has no attribute 'indirect_reference'
```

Reply:
Oh damn, I missed that part in my previous comment. Give me a second to re-introduce it :sweat_smile: